bubblewrap.c

```diff
@@ -379,6 +381,25 @@ handle_die_with_parent (void)
     die_with_error ("prctl");
 }
 
+static void
+gate_signals (int action, sigset_t *prevmask) // here
+{
+  sigset_t mask;
+
+  /* When unblocking, only restore if not previously blocked. */
+
+  sigemptyset (&mask);
+
+  if (action == SIG_BLOCK || !sigismember (prevmask, SIGINT))
+    sigaddset (&mask, SIGINT);
+
+  if (action == SIG_BLOCK || !sigismember (prevmask, SIGTERM))
+    sigaddset (&mask, SIGTERM);
```

Comment:
What's special about `SIGINT` and `SIGTERM` that means these signals need to be passed through, but for example `SIGQUIT` does not?

There's also some duplication here: this function knows which signals are being forwarded, but so does `monitor_child()`. I'd prefer it to look more like this:

```c
int forwarded_signals[] =
{
  SIGINT,
  SIGTERM,
  SIGCONT,
  SIGHUP,
  SIGQUIT,
  SIGUSR1,
  SIGUSR2,
  SIGWINCH,
};

static void
block_forwarded_signals (sigset_t *prevmask)
{
  size_t i;
...
  for (i = 0; i < N_ELEMENTS (forwarded_signals); i++)
    sigaddset (&mask, forwarded_signals[i]);
...
}

static int
monitor_child (...)
{
...
  sigaddset (&mask, SIGCHLD);

  for (i = 0; i < N_ELEMENTS (forwarded_signals); i++)
    sigaddset (&mask, forwarded_signals[i]);
...
}
```

The list of forwardable signals there is a bit of a guess. `flatpak-spawn` in https://github.com/flatpak/flatpak-xdg-utils also forwards `SIGTSTP`, but has to translate it into `SIGSTOP`, which seems odd:

```c
  /* We forward stop requests as real stop, because the default doesn't
     seem to be to stop for non-kernel sent TSTP??? */
  if (sig == SIGTSTP)
    sig = SIGSTOP;
```

so maybe it would be best to leave out `SIGTSTP` until/unless someone can explain why they need it.